### ISSUE TITLE
Create termination error message

### ISSUE THREAD

**Issue by Azolo** [Azolo/websockex#30]
When any of the `:gen` modules terminate the create an error log. WebSockex should do that too.

**Comment by jurta** [comment]
Yes, this is important because currently a WebSockex process exits silently in case of errors. Also please mention the `terminate/2` callback in documentation and examples to highlight the importance of error reporting in `terminate/2`.

**Comment by Azolo** [comment]
@jurta It shouldn't be completely silent. If there was an error or any other kind of non-normal exit then there would at least be an exit message that was printed. If it was an error in a callback then it should also have a stacktrace associated with it.

If that's not happening, then I would consider it a bug.

As for the documentation, PRs are welcome. 😺

**Comment by khodzha** [comment]
not creating new issue, but what's the proper way to disconnect from server?

**Comment by Azolo** [comment]
> not creating new issue, but what's the proper way to disconnect from server?

I think that is a really good documentation issue. But the proper way is to send a `close` message.

The easiest way to do this is probably with a [`:close` callback return from a cast][1].

[1]: https://hexdocs.pm/websockex/WebSockex.html#c:handle_cast/2
